assets/src/edit-story/components/panels/publish/index.js

```diff
@@ -0,0 +1,246 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useRef, useState } from 'react';
+import styled from 'styled-components';
+import { rgba } from 'polished';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Row, DropDown, DateTime, Label, Media, Required } from '../../form';
+import { SimplePanel } from '../panel';
+import useInspector from '../../inspector/useInspector';
+import { useStory } from '../../../app/story';
+import { ReactComponent as ToggleIcon } from '../../../icons/dropdown.svg';
+import { useKeyDownEffect } from '../../keyboard';
+import useFocusOut from '../../../utils/useFocusOut';
+import { getReadableDate, getReadableTime } from './utils';
+
+const LabelWrapper = styled.div`
+  width: 106px;
+`;
+
+const FieldLabel = styled(Label)`
+  flex-basis: ${({ width }) => (width ? width : '64px')};
+`;
+
+const MediaWrapper = styled.div`
+  flex-basis: 134px;
+`;
+
+const BoxedText = styled.div.attrs({ role: 'button', tabIndex: '0' })`
+  color: ${({ theme }) => theme.colors.fg.v1};
+  font-family: ${({ theme }) => theme.fonts.body2.family};
+  font-size: ${({ theme }) => theme.fonts.body2.size};
+  line-height: ${({ theme }) => theme.fonts.body2.lineHeight};
+  letter-spacing: ${({ theme }) => theme.fonts.body2.letterSpacing};
+  display: flex;
+  flex-direction: row;
+  background-color: ${({ theme }) => rgba(theme.colors.fg.v1, 0.1)};
+  flex: 1;
+  padding: 2px;
+  border-radius: 4px;
+  :focus {
+    outline: -webkit-focus-ring-color auto 5px;
+  }
+`;
+
+const DateWrapper = styled.div`
+  padding: 5px 0 5px 2px;
+`;
+
+const Date = styled.span`
+  color: ${({ theme }) => rgba(theme.colors.fg.v1, 0.86)};
+`;
+
+const Time = styled.span`
+  color: ${({ theme }) => rgba(theme.colors.fg.v1, 0.4)};
+`;
+
+const DateTimeWrapper = styled.div`
+  position: relative;
+`;
+
+const StyledToggleIcon = styled(ToggleIcon)`
+  height: 26px;
+  flex: 1;
+`;
+
+function PublishPanel() {
+  const {
+    state: { users },
+  } = useInspector();
+
+  const {
+    state: {
+      meta: { isSaving },
+      story: { author, date, featuredMediaUrl, publisherLogoUrl },
+      capabilities,
+    },
+    actions: { updateStory },
+  } = useStory();
+
+  const [showDatePicker, setShowDatePicker] = useState(false);
+  const dateTimePickerNode = useRef();
+  const dateFieldNode = useRef();
+
+  useKeyDownEffect(
+    dateFieldNode,
+    { key: ['space', 'enter'] },
+    () => {
+      setShowDatePicker(!showDatePicker);
```

Comment:
```suggestion
      setShowDatePicker((val) => !val);
```

Then you can remove `showDatePicker` from the deps array below.